pkg/config/config.go

```diff
@@ -1245,13 +1247,35 @@ func (c *Config) ActiveDestination() (uri, identity string, err error) {
 // If searchPATH is set to true it will also search in $PATH.
 func (c *Config) FindHelperBinary(name string, searchPATH bool) (string, error) {
 	dirList := c.Engine.HelperBinariesDir
+	var bindirPath string
+	var bindirErr error
 
 	// If set, search this directory first. This is used in testing.
 	if dir, found := os.LookupEnv("CONTAINERS_HELPER_BINARY_DIR"); found {
 		dirList = append([]string{dir}, dirList...)
 	}
 
 	for _, path := range dirList {
+		if path == bindirPrefix || strings.HasPrefix(path, bindirPrefix+string(filepath.Separator)) {
+			// Calculate the path to the executable first time we encounter a $BINDIR prefix.
+			if bindirPath == "" && bindirErr == nil {
+				podman, bindirErr := os.Executable()
```

Comment:
nit: This should not be called podman, since this library is used for other tools like buildah.

Just call it executable

Reply:
Sure that makes sense, I'm using `execPath` now.

(It's a bit odd though, since `config_darwin.go` in this same package uses `libexec/podman` for the helper directory, but I guess that's perhaps expected since the helpers are somewhat shared between both?)